### ISSUE TITLE
Cannot find module 'undefined'

### ISSUE THREAD

**Issue by StarpTech** [lerna/lerna#999]
Lerna throws an issue when building on windows ci `AppVeyor` on local Windows machine and travis it works.

Lerna: 2.1.2
NodeJs: v4.8.4
npm: 2.15.11

Link to broken build: https://ci.appveyor.com/project/StarpTech/hemera/build/1.0.892

```
> hemera@ setup C:\projects\hemera
> npm i && lerna bootstrap
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
added 911 packages in 32.17s
lerna info version 2.1.2
lerna info versioning independent
lerna info Bootstrapping 26 packages
lerna info lifecycle preinstall
lerna info Installing external dependencies
lerna ERR! execute Error: Command failed: npm install
lerna ERR! execute npm WARN invalid config access=undefined
lerna ERR! execute npm WARN invalid config https-proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config init-author-url=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config loglevel="notice"
lerna ERR! execute npm WARN invalid config proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN onload-script failed to require onload script undefined
lerna ERR! execute npm WARN onload-script Error: Cannot find module 'undefined'
lerna ERR! execute npm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)
lerna ERR! execute npm WARN onload-script     at Function.Module._load (module.js:276:25)
lerna ERR! execute npm WARN onload-script     at Module.require (module.js:353:17)
lerna ERR! execute npm WARN onload-script     at require (internal/module.js:12:17)
lerna ERR! execute npm WARN onload-script     at cb (C:\Program Files (x86)\nodejs\node_modules\npm\lib\npm.js:289:9)
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\npm.js:382:14
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\config\core.js:82:7
lerna ERR! execute npm WARN onload-script     at Array.forEach (native)
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\config\core.js:81:13
lerna ERR! execute npm WARN onload-script     at f (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\once\once.js:25:25)
lerna ERR! execute npm WARN onload-script  { [Error: Cannot find module 'undefined'] code: 'MODULE_NOT_FOUND' }
lerna ERR! execute npm WARN invalid config access=undefined
lerna ERR! execute npm WARN invalid config also=undefined
lerna ERR! execute npm WARN invalid config https-proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config init-author-url=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config only=undefined
lerna ERR! execute npm WARN invalid config proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN onload-script failed to require onload script undefined
lerna ERR! execute npm WARN onload-script Error: Cannot find module 'undefined'
lerna ERR! execute npm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)
lerna ERR! execute npm WARN onload-script     at Function.Module._load (module.js:276:25)
lerna ERR! execute npm WARN onload-script     at Module.require (module.js:353:17)
lerna ERR! execute npm WARN onload-script     at require (internal/module.js:12:17)
lerna ERR! execute npm WARN onload-script     at cb (C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\npm.js:225:11)
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\npm.js:356:16
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\config\core.js:81:7
lerna ERR! execute npm WARN onload-script     at Array.forEach (native)
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\config\core.js:80:13
lerna ERR! execute npm WARN onload-script     at f (C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\node_modules\once\once.js:25:25)
lerna ERR! execute npm WARN onload-script  { [Error: Cannot find module 'undefined']
lerna ERR! execute npm WARN onload-script   stack: 'Error: Cannot find module \'undefined\'\n    at Function.Module._resolveFilename (module.js:325:15)\n    at Function.Module._load (module.js:276:25)\n    at Module.require (module.js:353:17)\n    at require (internal/module.js:12:17)\n    at cb (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:225:11)\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:356:16\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:81:7\n    at Array.forEach (native)\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:80:13\n    at f (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\once\\once.js:25:25)',
lerna ERR! execute npm WARN onload-script   code: 'MODULE_NOT_FOUND' }
lerna ERR! execute npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning undefined: error:0906D06C:PEM routines:PEM_read_bio:no start line
lerna ERR! execute npm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.
lerna ERR! execute npm ERR! error:0906D06C:PEM routines:PEM_read_bio:no start line
lerna ERR! execute 
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2017-08-30T08_08_15_899Z-debug.log
lerna ERR! execute 
lerna ERR! execute     at C:\projects\hemera\node_modules\execa\index.js:236:11
lerna ERR! execute     at process._tickCallback (node.js:438:9)
lerna ERR! execute  callback with error
lerna ERR! execute  { [Error: Command failed: npm install
lerna ERR! execute npm WARN invalid config access=undefined
lerna ERR! execute npm WARN invalid config https-proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config init-author-url=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config loglevel="notice"
lerna ERR! execute npm WARN invalid config proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN onload-script failed to require onload script undefined
lerna ERR! execute npm WARN onload-script Error: Cannot find module 'undefined'
lerna ERR! execute npm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)
lerna ERR! execute npm WARN onload-script     at Function.Module._load (module.js:276:25)
lerna ERR! execute npm WARN onload-script     at Module.require (module.js:353:17)
lerna ERR! execute npm WARN onload-script     at require (internal/module.js:12:17)
lerna ERR! execute npm WARN onload-script     at cb (C:\Program Files (x86)\nodejs\node_modules\npm\lib\npm.js:289:9)
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\npm.js:382:14
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\config\core.js:82:7
lerna ERR! execute npm WARN onload-script     at Array.forEach (native)
lerna ERR! execute npm WARN onload-script     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\config\core.js:81:13
lerna ERR! execute npm WARN onload-script     at f (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\once\once.js:25:25)
lerna ERR! execute npm WARN onload-script  { [Error: Cannot find module 'undefined'] code: 'MODULE_NOT_FOUND' }
lerna ERR! execute npm WARN invalid config access=undefined
lerna ERR! execute npm WARN invalid config also=undefined
lerna ERR! execute npm WARN invalid config https-proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config init-author-url=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN invalid config only=undefined
lerna ERR! execute npm WARN invalid config proxy=undefined
lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
lerna ERR! execute npm WARN onload-script failed to require onload script undefined
lerna ERR! execute npm WARN onload-script Error: Cannot find module 'undefined'
lerna ERR! execute npm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)
lerna ERR! execute npm WARN onload-script     at Function.Module._load (module.js:276:25)
lerna ERR! execute npm WARN onload-script     at Module.require (module.js:353:17)
lerna ERR! execute npm WARN onload-script     at require (internal/module.js:12:17)
lerna ERR! execute npm WARN onload-script     at cb (C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\npm.js:225:11)
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\npm.js:356:16
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\config\core.js:81:7
lerna ERR! execute npm WARN onload-script     at Array.forEach (native)
lerna ERR! execute npm WARN onload-script     at C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\lib\config\core.js:80:13
lerna ERR! execute npm WARN onload-script     at f (C:\Users\appveyor\AppData\Roaming\npm\node_modules\npm\node_modules\once\once.js:25:25)
lerna ERR! execute npm WARN onload-script  { [Error: Cannot find module 'undefined']
lerna ERR! execute npm WARN onload-script   stack: 'Error: Cannot find module \'undefined\'\n    at Function.Module._resolveFilename (module.js:325:15)\n    at Function.Module._load (module.js:276:25)\n    at Module.require (module.js:353:17)\n    at require (internal/module.js:12:17)\n    at cb (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:225:11)\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:356:16\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:81:7\n    at Array.forEach (native)\n    at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:80:13\n    at f (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\once\\once.js:25:25)',
lerna ERR! execute npm WARN onload-script   code: 'MODULE_NOT_FOUND' }
lerna ERR! execute npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning undefined: error:0906D06C:PEM routines:PEM_read_bio:no start line
lerna ERR! execute npm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.
lerna ERR! execute npm ERR! error:0906D06C:PEM routines:PEM_read_bio:no start line
lerna ERR! execute 
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2017-08-30T08_08_15_899Z-debug.log
lerna ERR! execute ]
lerna ERR! execute   stack: 'Error: Command failed: npm install\nnpm WARN invalid config access=undefined\nnpm WARN invalid config https-proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config init-author-url=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config loglevel="notice"\nnpm WARN invalid config proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN onload-script failed to require onload script undefined\nnpm WARN onload-script Error: Cannot find module \'undefined\'\nnpm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)\nnpm WARN onload-script     at Function.Module._load (module.js:276:25)\nnpm WARN onload-script     at Module.require (module.js:353:17)\nnpm WARN onload-script     at require (internal/module.js:12:17)\nnpm WARN onload-script     at cb (C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\npm.js:289:9)\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\npm.js:382:14\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\config\\core.js:82:7\nnpm WARN onload-script     at Array.forEach (native)\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\config\\core.js:81:13\nnpm WARN onload-script     at f (C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\once\\once.js:25:25)\nnpm WARN onload-script  { [Error: Cannot find module \'undefined\'] code: \'MODULE_NOT_FOUND\' }\nnpm WARN invalid config access=undefined\nnpm WARN invalid config also=undefined\nnpm WARN invalid config https-proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config init-author-url=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config only=undefined\nnpm WARN invalid config proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN onload-script failed to require onload script undefined\nnpm WARN onload-script Error: Cannot find module \'undefined\'\nnpm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)\nnpm WARN onload-script     at Function.Module._load (module.js:276:25)\nnpm WARN onload-script     at Module.require (module.js:353:17)\nnpm WARN onload-script     at require (internal/module.js:12:17)\nnpm WARN onload-script     at cb (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:225:11)\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:356:16\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:81:7\nnpm WARN onload-script     at Array.forEach (native)\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:80:13\nnpm WARN onload-script     at f (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\once\\once.js:25:25)\nnpm WARN onload-script  { [Error: Cannot find module \'undefined\']\nnpm WARN onload-script   stack: \'Error: Cannot find module \\\'undefined\\\'\\n    at Function.Module._resolveFilename (module.js:325:15)\\n    at Function.Module._load (module.js:276:25)\\n    at Module.require (module.js:353:17)\\n    at require (internal/module.js:12:17)\\n    at cb (C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\npm.js:225:11)\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\npm.js:356:16\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\config\\\\core.js:81:7\\n    at Array.forEach (native)\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\config\\\\core.js:80:13\\n    at f (C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\node_modules\\\\once\\\\once.js:25:25)\',\nnpm WARN onload-script   code: \'MODULE_NOT_FOUND\' }\nnpm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning undefined: error:0906D06C:PEM routines:PEM_read_bio:no start line\nnpm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.\nnpm ERR! error:0906D06C:PEM routines:PEM_read_bio:no start line\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\appveyor\\AppData\\Roaming\\npm-cache\\_logs\\2017-08-30T08_08_15_899Z-debug.log\n\n    at C:\\projects\\hemera\\node_modules\\execa\\index.js:236:11\n    at process._tickCallback (node.js:438:9)',
lerna ERR! execute   code: 1,
lerna ERR! execute   killed: false,
lerna ERR! execute   stdout: '',
lerna ERR! execute   stderr: 'npm WARN invalid config access=undefined\nnpm WARN invalid config https-proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config init-author-url=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config loglevel="notice"\nnpm WARN invalid config proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN onload-script failed to require onload script undefined\nnpm WARN onload-script Error: Cannot find module \'undefined\'\nnpm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)\nnpm WARN onload-script     at Function.Module._load (module.js:276:25)\nnpm WARN onload-script     at Module.require (module.js:353:17)\nnpm WARN onload-script     at require (internal/module.js:12:17)\nnpm WARN onload-script     at cb (C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\npm.js:289:9)\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\npm.js:382:14\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\config\\core.js:82:7\nnpm WARN onload-script     at Array.forEach (native)\nnpm WARN onload-script     at C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\lib\\config\\core.js:81:13\nnpm WARN onload-script     at f (C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\once\\once.js:25:25)\nnpm WARN onload-script  { [Error: Cannot find module \'undefined\'] code: \'MODULE_NOT_FOUND\' }\nnpm WARN invalid config access=undefined\nnpm WARN invalid config also=undefined\nnpm WARN invalid config https-proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config init-author-url=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN invalid config only=undefined\nnpm WARN invalid config proxy=undefined\nnpm WARN invalid config Must be a full url with \'http://\'\nnpm WARN onload-script failed to require onload script undefined\nnpm WARN onload-script Error: Cannot find module \'undefined\'\nnpm WARN onload-script     at Function.Module._resolveFilename (module.js:325:15)\nnpm WARN onload-script     at Function.Module._load (module.js:276:25)\nnpm WARN onload-script     at Module.require (module.js:353:17)\nnpm WARN onload-script     at require (internal/module.js:12:17)\nnpm WARN onload-script     at cb (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:225:11)\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\npm.js:356:16\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:81:7\nnpm WARN onload-script     at Array.forEach (native)\nnpm WARN onload-script     at C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\lib\\config\\core.js:80:13\nnpm WARN onload-script     at f (C:\\Users\\appveyor\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\once\\once.js:25:25)\nnpm WARN onload-script  { [Error: Cannot find module \'undefined\']\nnpm WARN onload-script   stack: \'Error: Cannot find module \\\'undefined\\\'\\n    at Function.Module._resolveFilename (module.js:325:15)\\n    at Function.Module._load (module.js:276:25)\\n    at Module.require (module.js:353:17)\\n    at require (internal/module.js:12:17)\\n    at cb (C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\npm.js:225:11)\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\npm.js:356:16\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\config\\\\core.js:81:7\\n    at Array.forEach (native)\\n    at C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\lib\\\\config\\\\core.js:80:13\\n    at f (C:\\\\Users\\\\appveyor\\\\AppData\\\\Roaming\\\\npm\\\\node_modules\\\\npm\\\\node_modules\\\\once\\\\once.js:25:25)\',\nnpm WARN onload-script   code: \'MODULE_NOT_FOUND\' }\nnpm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning undefined: error:0906D06C:PEM routines:PEM_read_bio:no start line\nnpm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.\nnpm ERR! error:0906D06C:PEM routines:PEM_read_bio:no start line\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\appveyor\\AppData\\Roaming\\npm-cache\\_logs\\2017-08-30T08_08_15_899Z-debug.log\n',
lerna ERR! execute   failed: true,
lerna ERR! execute   signal: null,
lerna ERR! execute   cmd: 'npm install',
lerna ERR! execute   timedOut: false }
lerna WARN complete Waiting for 3 child processes to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hemera@ setup: `npm i && lerna bootstrap`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the hemera@ setup script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2017-08-30T08_08_24_525Z-debug.log
Command exited with code 1
```

**Comment by StarpTech** [comment]
Looks like that the current broken build of lerna on Node 4 https://ci.appveyor.com/project/lerna/lerna/branch/master/job/yaot27e7vtv2b28g has the same issue.

**Comment by StarpTech** [comment]
It also fails on Node 6 :/

**Comment by evocateur** [comment]
This has been vexing me, and I don't have a local Windows env to experiment in. I _suspect_ this is an environmental difference in making SSL requests in node v4, judging by the PEM errors in the logs.

Really looking forward to dropping support for node v4, btw.

**Comment by Zalastax** [comment]
I'm having the same problem (same errors about invalid config x=undefined).  I've tried both yarn and npm, and running under both an admin account and a normal account. I'll do further investigation on my system if you have any ideas.

Errors come from performing bootstrap on a fresh clone of: https://github.com/gatsbyjs/gatsby/commit/d1fa4b867a081786f1f6c27ffdbd8dc79e997aeb

OS: Windows 10
Lerna: 2.1.1 (and 2.1.2 if I bump the gatsby project, same error)
NodeJs: v8.4.0
yarn: 0.28.4
npm: 5.3.0

**Comment by paulelsner** [comment]
Same issue here:
Node v8.4.0
lerna info version 2.1.2

> build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config access=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config also=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config https-proxy=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config init-author-url=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config only=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config proxy=undefined
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN invalid config Must be a full url with 'http://'
build	05-Sep-2017 00:26:51	         lerna ERR! execute npm WARN onload-script failed to require onload script undefined

**Comment by hoodie** [comment]
This might be somehow related: https://github.com/nodejs/node/issues/14593 (fun read)
TL;DR: ConEmu injects a some DLL that overwrites some WinAPI calles. This seems to triggere this behaviour (perhaps somewhere in [here](https://github.com/Maximus5/ConEmu/blob/1b6bb22e8be3b0d29b10f23abfdd31aed44a756f/src/ConEmuHk/hkEnvironment.h)).

Now I don't expect AppVeyor runs ConEmu, but perhaps it does something similar.

**Comment by evocateur** [comment]
That's a fun thread. We're _probably_ experiencing this as a side-effect of https://github.com/sindresorhus/execa/pull/88, which we picked up in a recent dependency bump. It's weird, because most of the time you _do_ want the `env` to be propagated to the child, so I figured it was an okay change.

If we pass `extendEnv: false` (ugh) in the execa opts, I imagine we can probably avoid this?

**Comment by Zalastax** [comment]
Problem solved for me! @hoodie you're awesome!
I was running `lerna bootstrap` in a ConEmu shell and switching to cmd.exe makes the script run fine.

**Comment by StarpTech** [comment]
I can still reproduce it with the latest lerna version on Appveyor CI. 
https://ci.appveyor.com/project/StarpTech/bootme/build/job/kywnh6ygeeagchnq

**Comment by Zalastax** [comment]
@StarpTech does @evocateur's idea solve the problem? I personally switched to not call lerna from a ConEmu shell.

**Comment by StarpTech** [comment]
@Zalastax It's on CI server. They don't use conemu

**Comment by StarpTech** [comment]
@Zalastax How can I pass them?

**Comment by Zalastax** [comment]
[These are the places using execa](https://github.com/lerna/lerna/search?utf8=%E2%9C%93&q=execa&type=). Add to the opts object `extendEnv: false` as necessary. I guess it's fine to add it everywhere when you test and then we can refine later. After you've done that open a pull request with your changes and AppVeyor will try to build it.

**Comment by StarpTech** [comment]
Thanks for the instructions I will try it ASAP.

**Comment by CodeTroopers** [comment]
Can `extendEnv: false` be passed on the command line : `lerna bootstrap --extendEnv=false`?

**Comment by evocateur** [comment]
No, that’s a library option

**Comment by matryoshcow** [comment]
Same issue here when trying to roll a `commitizen` setup (https://www.npmjs.com/package/commitizen) & `ghooks` (https://www.npmjs.com/package/ghooks) with a *pre-commit* hook that runs lerna.

OS: Windows 10
Lerna: 2.5.1
Node: 8.9.0
npm: 5.51

**Comment by payload** [comment]
Again, this is most likely an issue in nodejs itself. Please raise awareness to this problem in this issue nodejs/node#14593

**Comment by CodeTroopers** [comment]
@payload Really!?? I have the same issue [only under NodeJS 4 with Appveyor](https://ci.appveyor.com/project/JaKXz/istanbuljs/build/208) and not with NodeJS 6. And not more with NodeJS 4, NodeJS 6 or NodeJS 8 on my local environment

**Comment by matryoshcow** [comment]
The issue went away for me upon updating [cmder](https://github.com/cmderdev/cmder/releases/) (the console emulator that I'm using). At the time of this writing the latest [cmder](https://github.com/cmderdev/cmder/releases/) version is 1.3.4.
